posthog/client.py

```
@@ -849,6 +859,21 @@ def get_feature_flag_payload(
 
         return payload
 
+    def get_decrypted_feature_flag_payload(self, key: str):
+        if self.disabled:
+            return None
+
+        if self.personal_api_key is None:
+            self.log.warning(
+                "[FEATURE FLAGS] You have to specify a personal_api_key to fetch decrypted feature flag payloads."
+            )
+            return None
+
+        try:
+            return self.get_remote_config_payload(key)
+        except Exception as e:
+            self.log.exception(f"[FEATURE FLAGS] Unable to get decrypted feature flag payload: {e}")
```

Feel free to disregard, but I wonder if we should be capturing all exceptions here. For example, if a personal api key is supplied, but incorrect, I'd almost want my app to crash with a message to that effect. It would. make the issue immediately noticeable with a clear path to fix it. Otherwise, this call fails silently and I have to check the logs to understand why.

---

Just for context, I'm on the fence on this. I'm trying to decide how to approach this with `posthog-dotnet`. I can see the other scenario where you have a working app both locally and remote. You add this one API call and it works locally, but breaks in production because you forgot to deploy the `personal_api_key` to production. 

In that scenario, perhaps you want the app to keep running and just log the error because you'll see your new call isn't working and you can then check the logs to figure out why.

---

At minimum, I think we may want to specifically handle the exception when we get a 401 status and show the detail. It'll say something like "Personal API key found in request Authorization header is invalid." which is much more helpful when trying to figure out why it's not working.

---

Disregard my last message. I see that we are showing the exception message as part of the logged message. I need to make sure to do that in my own code. 😆

---

Hmm, yeah that is tough. I agree that we should probably be propagating up certain types of errors. I also feel like as a developer I might only expect the SDK to provide detailed / helpful error messages to stdout, but would not expect to have to try/catch every SDK call just in case. 

I did it this way as it seemed to be the established pattern, and I think we should probably stick with it until having a conversation with the larger SDK group about it!

---

Yup! Sounds good to me.

---

fwiw, in C# I tend to do something like this:

```csharp
try {}
catch (Exception e) when (e is not ArgumentException and not NullReferenceException) {}
```

Mainly because `ArgumentException` and `NullReferenceException` are 99.99% of the time are due to a deterministic coding mistake during development and not something anyone would encounter in production unless I deployed a coding mistake. Those I want immediately apparent during development. I don't expect consumers to try/catch those because I make every effort to ensure they don't happen.